blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
no, I don't thinks so. I agree that the fix was wrong and the root cause might be the variable _branches in HttpTxManager and branches_ in XAResourceManager look like not sync. when XAResourceManager clean the branches and there were still the references in the HttpTxManager. I just wonder that why we need to have these two ?